### ISSUE TITLE
tlog : get_last_data example is not working anymore

### ISSUE THREAD

**Issue by iwanbk** [g8os/blockstor#81]
```
$ ./get_last_data 
2017/04/28 09:44:19 k=4, m=2
data - len = 5101
data - len = 5101
data - len = 5101
data - len = 5101
2017/04/28 09:44:19 unc len = 60res = 2
2017/04/28 09:44:19 decode capnp
panic: capnp: address out of bounds

goroutine 1 [running]:
github.com/g8os/blockstor/vendor/zombiezen.com/go/capnproto2.List.primitiveElem(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x100000000, 0x0, 0xc420055968, 0x4ee2f5)
        /home/ibk/pathgo/src/github.com/g8os/blockstor/vendor/zombiezen.com/go/capnproto2/list.go:184 +0x179
github.com/g8os/blockstor/vendor/zombiezen.com/go/capnproto2.PointerList.PtrAt(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/ibk/pathgo/src/github.com/g8os/blockstor/vendor/zombiezen.com/go/capnproto2/list.go:310 +0xb4
github.com/g8os/blockstor/vendor/zombiezen.com/go/capnproto2.(*Message).RootPtr(0xc4200a02d0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc420055b48, 0x4f25f0)
        /home/ibk/pathgo/src/github.com/g8os/blockstor/vendor/zombiezen.com/go/capnproto2/mem.go:121 +0x14d
github.com/g8os/blockstor/tlog/schema.ReadRootTlogAggregation(0xc4200a02d0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/ibk/pathgo/src/github.com/g8os/blockstor/tlog/schema/tlog_schema.capnp.go:233 +0x7f
main.decodeCapnp(0x84f900, 0xc420020690)
        /home/ibk/pathgo/src/github.com/g8os/blockstor/tlog/tlogclient/examples/get_last_data/main.go:154 +0x1b3
main.main()
        /home/ibk/pathgo/src/github.com/g8os/blockstor/tlog/tlogclient/examples/get_last_data/main.go:93 +0x660
```

Looks like caused by the change in encryption

**Comment by iwanbk** [comment]
i've modified the code to decrypt to:
```
func decrypt(encrypted []byte, privKey, nonce string) []byte {
    decrypter, err := tlog.NewAESDecrypter(privKey, nonce)
    if err != nil {
        log.Fatalf("failed to create aes decrypter:%v", err)
    }

    plain, err := decrypter.Decrypt(encrypted)
    if err != nil {
        log.Fatalf("failed to decrypt:%v", err)
    }
    return plain
}
```
got this error: 
```
failed to decrypt:cipher: message authentication failed
```

@GlenDC can you please check if i use that class correctly?

**Comment by GlenDC** [comment]
@iwanbk looks correct to me, and the AESDecrypter and AESEncrypter have been tested, to ensure that whats get encrypted by it, can also be decrypted again. So there must be something wrong with the cipher text (`encrypted []byte`) that you feed into it.

By the way you can use "\`\`\`go" instead of "\`\`\`" to provide syntax high lighting in code blocks when using markdown. It's not available in all markdowns, but it is in the one that github uses.

**Comment by iwanbk** [comment]
```
 and the AESDecrypter and AESEncrypter have been tested, to ensure that whats get encrypted by it, can also be decrypted again
```
yep, i saw the test.
```
So there must be something wrong with the cipher text (encrypted []byte) that you feed into it.
```
this failure started at f481c309f048c5ef656ccb2aac18437f6b2b28f5 and there were no changes in get_last_data at that commit.
So i think the issue might be something else.

**Comment by iwanbk** [comment]
```
So there must be something wrong with the cipher text (encrypted []byte) that you feed into it
```
checked it, the length is differ by one byte

**Comment by iwanbk** [comment]
Fixed in bec650984b6279981a9e43b9b18ffc5c3673fb5d by align the data before encrypt it.
more description can be found in the commit message.

**Comment by iwanbk** [comment]
Fixed in #84
